extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -24,76 +25,192 @@
-    @NotNull
+    private static final String DATA_KEY = "data";
+    private static final String RENEWABLE_KEY = "renewable";
+    private static final String AUTH_KEY = "auth";
+    private static final String LEASE_DURATION_KEY = "lease_duration";
+    private static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE = new TypeReference<>() {
+    };
+    private static final String INCREMENT_SECONDS_FORMAT = "%ds";
+    private static final String INCREMENT_KEY = "increment";
+
     private final EdcHttpClient httpClient;
-    @NotNull
+    private final Headers headers;
     private final ObjectMapper objectMapper;
+    private final HashicorpVaultConfigValues configValues;
+    private final HttpUrl healthCheckUrl;
+    private final Monitor monitor;
 
-    HashicorpVaultClient(@NotNull HashicorpVaultClientConfig hashicorpVaultConfig, @NotNull EdcHttpClient httpClient,
-                         @NotNull ObjectMapper objectMapper) {
-        this.hashicorpVaultConfig = hashicorpVaultConfig;
+    HashicorpVaultClient(@NotNull EdcHttpClient httpClient,
+                         @NotNull ObjectMapper objectMapper,
+                         @NotNull Monitor monitor,
+                         @NotNull HashicorpVaultConfigValues configValues) {
         this.httpClient = httpClient;
         this.objectMapper = objectMapper;
+        this.monitor = monitor;
+        this.configValues = configValues;
+        this.headers = getHeaders();
+        this.healthCheckUrl = getHealthCheckUrl();
+    }
+
+    public Result<Void> doHealthCheck() {
+        var request = httpGet(healthCheckUrl);
+
+        try (var response = httpClient.execute(request)) {
+            if (response.isSuccessful()) {
+                return Result.success();
+            }
+
+            var code = response.code();
+            var errMsg = switch (code) {
+                case 429 -> "Vault is in standby";
+                case 472 -> "Vault is in recovery mode";
+                case 473 -> "Vault is in performance standby";
+                case 501 -> "Vault is not initialized";
+                case 503 -> "Vault is sealed";
+                default -> "Vault returned unspecified code %d".formatted(code);
+            };
+            var body = response.body();
+            if (body == null) {
+                return Result.failure("Healthcheck returned empty response body");
+            }
+            return Result.failure("Vault is not available. Reason: %s, additional information: %s".formatted(errMsg, body.string()));
+        } catch (IOException e) {
+            return Result.failure("Failed to perform healthcheck with reason: %s".formatted(e.getMessage()));
+        }
+    }
+
+    /**
+     * Attempts to look up the Vault token defined in the configurations.
+     * <p>
+     * Will retry in some error cases.
+     *
+     * @return boolean indicating if the token is renewable
+     */
+    public Result<Boolean> lookUpToken() {
```

Comment:
minor nit: if this method's purpose is to look up, whether the token is renewable, then it should be called `isTokenRenewable` or `checkTokenRenewable`. A name `lookUpToken` may leave people puzzled as to why this returns a boolean and not a token.